### DISCUSSION TITLE
Enhance Focus Positions for Live Collaboration

### DISCUSSION THREAD

**Discussion by jjvraw** [mdx-editor/editor#117] [Ideas]
Expanding the capability of autoFocus beyond just `rootStart` or `rootEnd` to improve cursor management would be greatly beneficial especially during live, socket-handled, collaboration sessions within the editor. 

It also appears that there isn't a straightforward way to get a direct reference to the text space within the editor, which makes it challenging to set the cursor position programmatically.

**Comment by petyosi** [comment]
Live collaboration is not something in my short/mid-term priorities. I'm vaguely aware that Lexical has a plugin for that purpose, but I have not explored it. This being said, I'm happy to review a contribution in that direction. 

The second paragraph does not make sense to me, I'm afraid. What's text space?

**Reply by jjvraw** [reply]
I understand. With that being said, would you consider extending [MDXEditorMethods interface](https://mdxeditor.dev/editor/api/editor.mdxeditormethods)'s `focus` property to accept some form of position? or are you considering something more complete? 

Regarding the second paragraph, I was referring to a method of obtaining a direct JavaScript reference/`ref`(to the underlying LexicalEditor instance when using the MDXEditor. To allow us to set the cursor position, manipulate selections, or perform other advanced operations by directly interacting with the LexicalEditor API.

**Reply by petyosi** [reply]
I believe that `focus` is a [re-export of the lexical editor method](https://github.com/facebook/lexical/blob/main/packages/lexical/src/LexicalEditor.ts#L1093) and its options.

The way I see the editor extended is through the plugins API. Programming through there gives you quite the flexibility - including access to the internal MDX Editor state, for example. You also get access to the evegreen values like the current editor instance (there can be nested ones) and the current selection. 

Take a look at the source code of some of the plugins - the link dialog one is a good example of several basic concepts. The node based state management might look a bit intimidating, but this is just a first impression.

**Reply by jjvraw** [reply]
Okay, thank you for the insight. I assume you would then prefer to keep the currently functionality of `focus` to behave as a re-export and rather leverage the plugins API to extend the editor's capabilities? or is there still interest in adding cursor/caret management functionality as an MDXEditor method?

**Reply by unidesigner** [reply]
Hi @jjvraw ! - I came across this discussion because I have the same use case. I'd like to retrieve the current cursor position from the editor, and set the cursor position position programatically after a setMarkdown function call. Have you found a workaround without modifying the focus method? Thanks!
